### PR TITLE
fix: Parse JSON for updateSettings

### PR DESCRIPTION
**Pull request by dpgraham**
Need to parse 'settings' JSON before passing it through to wd

(cc: @wswebcreation)

### CODE REVIEW

File: app/renderer/components/Inspector/Actions.js [hunk 1]
```diff
@@ -51,6 +51,20 @@ export default class Actions extends Component {
       }
     }
 
+    // Special case for 'updateSettings'
+    if (action.methodName === 'updateSettings') {
+      if (_.isString(args[0])) {
```
Comment (mykola-mokhnach) [inline app/renderer/components/Inspector/Actions.js]: this if could be combined with the previous one using &&

File: app/renderer/components/Inspector/Actions.js [hunk 2]
```diff
@@ -51,6 +51,20 @@ export default class Actions extends Component {
...
+        try {
+          args[0] = JSON.parse(args[0]);
+        } catch (e) {
+          notification.error({
+            message: t('invalidJson', {json: args[1]}),
```
Comment (mykola-mokhnach) [inline app/renderer/components/Inspector/Actions.js]: do we know that we have at least 2 args?

Comment (mykola-mokhnach) [reply app/renderer/components/Inspector/Actions.js]: or it should be args[0]?

Comment (dpgraham) [reply app/renderer/components/Inspector/Actions.js]: Good catch, thanks.

**Comment by wswebcreation** [comment]
Hi @dpgraham 

Thanks for this fix. Is this already in the latest release? I tried this with the 1.12.0 and it still fails

**Pull request merged**